tests/ftqc/test_tape_prototype.py

```diff
@@ -129,18 +157,314 @@ def circ():
     assert np.allclose(res, expected_res, atol=0.05)
 
 
-@pytest.mark.parametrize(
-    "backend, expected_method",
-    [(LightningQubitBackend(), "one-shot"), (NullQubitBackend(), "device")],
-)
-@pytest.mark.parametrize(
-    "initial_config", [ExecutionConfig(mcm_config=MCMConfig(mcm_method="placeholder")), None]
-)
-def test_setup_execution_config(backend, expected_method, initial_config):
-    """Test that setup_execution_config works as expected"""
+class TestQuantumScriptSequence:
+    """Test the behaviour of the class introduced to represent a QuantumScript
+    that is broken into multipe execution steps (potentially with additional runtime
+    behaviour between each segment)"""
 
-    dev = FTQCQubit(wires=2, backend=backend)
-    config = dev.setup_execution_config(initial_config)
+    def get_test_sequence(self):
+        operations = [[X(0), Y(1)], [Y(0), Z(0)], [H(1), S(0)]]
+        measurements = [[], [], [qml.expval(X(0))]]
+        tapes = [
+            qml.tape.QuantumScript(ops, measurements=meas, shots=Shots(1000))
+            for ops, meas in zip(operations, measurements)
+        ]
+        sequence = QuantumScriptSequence(tapes)
+        return sequence
 
-    assert isinstance(config, ExecutionConfig)
-    assert config.mcm_config.mcm_method == expected_method
+    @pytest.mark.parametrize("shots", [None, Shots([10, 10])])
+    def test_quantum_sequence_initializes(self, shots):
+        """Test that a QuantumScriptSequence can be initialized as expected"""
+
+        ops1 = [X(0), Y(1), qml.RX(1.23, 2)]
+        ops2 = [qml.RY(1.2, 0), RZ(0.45, 0), X(1)]
+        ops3 = [S(2), H(1), qml.CNOT([1, 0]), S(0)]
+        final_measurements = [qml.expval(X(0)), qml.sample(wires=[2])]
+
+        operations = [ops1, ops2, ops3]
+        measurements = [[], [], final_measurements]
+
+        tapes = [
+            qml.tape.QuantumScript(ops, measurements=meas, shots=Shots(1000))
+            for ops, meas in zip(operations, measurements)
+        ]
+
+        sequence = QuantumScriptSequence(tapes, shots=shots)
+
+        # wires are as expected
+        assert sequence.num_wires == 3
+        assert sequence.wires == qml.wires.Wires([0, 1, 2])
+
+        # shots are as expected
+        assert sequence.shots == Shots(1000) if shots is None else shots
+        for tape in sequence.tapes:
+            assert tape.shots == Shots(1)
+
+        # stored tapes are as expected
+        assert len(sequence.tapes) == 3
+        assert len(sequence.intermediate_tapes) == 2
+        assert isinstance(sequence.final_tape, qml.tape.QuantumScript)
+
+        # overall measurements and operations are as expected
```

Comment:
Just wondering if it is also worth defining iteration rules for this type, and ensuring the correct order is observed (ie defining `__iter__` / `__next__` so the order of iteration is always 1-directional to match expectations?)